llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp

```diff
@@ -11841,9 +11841,9 @@ void BoUpSLP::reorderInputsAccordingToOpcode(ArrayRef<Value *> VL,
 }
 
 Instruction &BoUpSLP::getLastInstructionInBundle(const TreeEntry *E) {
-  auto &Res = EntryToLastInstruction.FindAndConstruct(E);
-  if (Res.second)
-    return *Res.second;
+  auto &Res = EntryToLastInstruction[E];
```

Comment:
I would prefer find instead

Reply:
`DenseMap::find` won't default-construct.  Would you be OK with something like this?

```
auto It = EntryToLastInstruction.find(E);
if (It != EntryToLastInstruction.end())
  return *It->second;
auto &Res = EntryToLastInstruction.insert(E).first->second;
```

Reply:
Yes

Reply:
The latest iteration addresses the issues.